src/main/java/org/folio/circulation/domain/notice/schedule/ScheduledDigitalReminderHandler.java

```diff
@@ -206,22 +248,40 @@ private CompletableFuture<Result<ScheduledNoticeContext>> createFeeFineAction(Sc
    */
   @Override
   protected CompletableFuture<Result<ScheduledNotice>> updateNotice(ScheduledNoticeContext context) {
-    OverdueFinePolicyRemindersPolicy.ReminderSequenceEntry nextReminder = context.getLoan().getNextReminder();
+    RemindersPolicy.ReminderConfig nextReminder = context.getLoan().getNextReminder();
     if (nextReminder == null) {
       return deleteNotice(context.getNotice(), "no more reminders scheduled");
     } else if (isNoticeIrrelevant(context)) {
       return deleteNotice(context.getNotice(), "further reminder notices became irrelevant");
     } else {
-      ScheduledNotice nextReminderNotice = context.getNotice()
-        .withNextRunTime(nextReminder.getPeriod().plusDate(systemTime));
-      nextReminderNotice.getConfiguration()
-        .setTemplateId(nextReminder.getNoticeTemplateId())
-        .setFormat(nextReminder.getNoticeFormat());
-
-      return scheduledNoticesRepository.update(nextReminderNotice);
+      return buildNextNotice(context, nextReminder)
+        .thenCompose(scheduledNoticeResult -> scheduledNoticesRepository.update(scheduledNoticeResult.value()));
     }
   }
 
+  protected CompletableFuture<Result<ScheduledNotice>> buildNextNotice (
+    ScheduledNoticeContext context, RemindersPolicy.ReminderConfig nextReminder) {
+
+    return configurationRepository.findTimeZoneConfiguration()
+      .thenCompose(tenantTimeZone ->
+        nextReminder
+          .nextNoticeDueOn(
+            systemTime,
+            tenantTimeZone.value(),
+            context.getLoan().getCheckoutServicePointId(),
+            calendarRepository
+          )
+          .thenCompose(nextRunTimeResult -> {
+            ScheduledNotice nextReminderNotice = context.getNotice()
+              .withNextRunTime(nextRunTimeResult.value().truncatedTo(ChronoUnit.HOURS));
+            nextReminderNotice.getConfiguration()
+              .setTemplateId(nextReminder.getNoticeTemplateId())
+              .setFormat(nextReminder.getNoticeFormat());
+            return ofAsync(nextReminderNotice);
+          })
```

Comment:
It looks like this is composed to the nextNoticeDueOn result, not to the previous thenCompose, which makes the whole thing look more complex. Is this necessary?
Also, instead of inlining code with { } it would make sense to extract a method here.